### PR TITLE
Add getCurrentGeofences() to track monitored geofences

### PR DESCRIPTION
**Pull request by evan-masseau**
# Description
Add `getCurrentGeofences()` method to the LocationManager interface to distinguish which geofences are currently being tracked by the system's GeofencingClient vs just stored locally. Uses a separate list of tracked IDs persisted to dataStore.

## Due Diligence
- [x] I have tested this on an emulator and/or a physical device.
- [x] I have added sufficient unit/integration tests of my changes.
- [ ] I have adjusted or added new test cases to team test docs, if applicable.
- [x] I am confident these changes are compatible with all Android versions the SDK currently supports.

## Release/Versioning Considerations
- [ ] `Patch` Contains internal changes or backwards-compatible bug fixes.
- [x] `Minor` Contains changes to the public API. 
- [ ] `Major` Contains **breaking** changes.
- [ ] Contains readme or migration guide changes.
- [x] This is planned work for an upcoming release.

### SDK Changes
- Add `getCurrentGeofences(): List<KlaviyoGeofence>` to `LocationManager` interface
- Store tracked geofence IDs separately in `klaviyo_tracked_geofence_ids` dataStore key
- Save tracked IDs on successful `addGeofences()` callback
- Clear tracked IDs on successful `removeGeofences()` callback
- Add helper methods: `storeTrackedIds()`, `getTrackedIds()`, `clearTrackedIds()`

### Test Changes
- Add 7 unit tests for `getCurrentGeofences()` functionality
- Fix mock initialization order bug (`mockPendingIntent` declared before `mockGeofencingClient`)
- Update `mockRemoveGeofencesTask` to auto-invoke success callback

### Sample App Changes
- Use `getCurrentGeofences()` instead of `getStoredGeofences()` to display only monitored fences
- Add refresh button to map view as fallback for timing edge cases

## Test Plan
1. Run unit tests: `./gradlew :sdk:location:testDebugUnitTest`
2. Install sample app: `./gradlew :sample:installDebug`
3. Grant location permissions and register for geofencing
4. Verify map shows only currently monitored geofences (max 20)
5. Tap refresh button to manually re-fetch current geofences

## Related Issues/Tickets
Part of Flutter support feature branch (`feat/flutter-support`)

### CODE REVIEW

File: sample/src/main/java/com/klaviyo/sample/SampleViewModel.kt [hunk 1]
```diff
@@ -176,16 +185,16 @@ class SampleViewModel : ViewModel() {
...
 
         if (hasPermission) {
-            // Get current geofences from the location module, and then tap in to the internal geofence sync events subscription to keep the view updated.
+            // Get currently monitored geofences from the location module, and then tap in to the internal geofence sync events subscription to keep the view updated.
             // Note: this uses advanced APIs used for demonstration, not necessary for a typical integration
             Registry.getOrNull<LocationManager>()?.apply {
-                monitoredGeofences = getStoredGeofences()
-                onGeofenceSync(true, ::setGeofences)
+                monitoredGeofences = getCurrentGeofences()
+                onGeofenceSync(true, ::syncGeofences)
```
[the remark this answers is not in the document]
Comment (evan-masseau) [reply sample/src/main/java/com/klaviyo/sample/SampleViewModel.kt]: Yeah, not anymore, because the list that `onGeofenceSync` sends is the list fetched from the API, but for the sample app purpose we are just showing the _monitored_ fences. So I'm really just using `onGeofenceSync` as a proactive hook to check the actual _monitored_ fences.

File: sdk/location/src/main/java/com/klaviyo/location/LocationManager.kt [hunk 2]
```diff
@@ -63,4 +63,15 @@ interface LocationManager {
...
+    /**
+     * Get the list of geofences currently being monitored by the system
+     *
+     * This returns only geofences that have been successfully registered with
+     * the system's GeofencingClient, which may be a subset of [getStoredGeofences]
+     * if filtering was applied (e.g., nearest 20 based on user location).
+     *
+     * @return List of geofences actively being monitored, or empty if none
+     */
+    fun getCurrentGeofences(): List<KlaviyoGeofence>
```
[the remark this answers is not in the document]
Comment (evan-masseau) [reply sdk/location/src/main/java/com/klaviyo/location/LocationManager.kt]: It isn't, but considering this is marked as an internal API I'm ok with that difference. To access this method you'd go through `core.Registry` to get the location service, `Registry.get<LocationManager>()`, sorta android's version of when we mark iOS methods `@spi`

[the remark this answers is not in the document]
Comment (belleklaviyo) [reply sdk/location/src/main/java/com/klaviyo/location/LocationManager.kt]: got it, okay thanks for clarifying! yeah that different is ok w me
